modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -342,79 +341,83 @@ impl ServerModule for Wallet {
+
+                let old_consensus_height = self.consensus_block_height(dbtx).await;
+
+                dbtx.insert_entry(&BlockHeightVoteKey(peer_id), &block_height)
+                    .await;
+
+                let new_consensus_height = self.consensus_block_height(dbtx).await;
+
+                if new_consensus_height > old_consensus_height {
+                    self.sync_up_to_consensus_height(
+                        dbtx,
+                        old_consensus_height,
+                        new_consensus_height,
+                    )
+                    .await;
                 }
-                WalletConsensusItem::PegOutSignature(peg_out_signature) => {
-                    let txid = peg_out_signature.txid;
-                    let unsigned_key = UnsignedTransactionKey(txid);
-
-                    if let Some(mut unsigned) = dbtx.get_value(&unsigned_key).await {
-                        if self
-                            .sign_peg_out_psbt(&mut unsigned.psbt, &peer_id, &peg_out_signature)
-                            .is_ok()
-                        {
-                            dbtx.insert_entry(&unsigned_key, &unsigned).await;
-
-                            if let Ok(pending_tx) = self.finalize_peg_out_psbt(unsigned) {
-                                // We were able to finalize the transaction, so we will delete the
-                                // PSBT and instead keep the
-                                // extracted tx for periodic transmission
-                                // and to accept the change into our wallet eventually once
-                                // it confirms.
-                                dbtx.insert_new_entry(&PendingTransactionKey(txid), &pending_tx)
-                                    .await;
-                                dbtx.remove_entry(&PegOutTxSignatureCI(txid)).await;
-                                dbtx.remove_entry(&unsigned_key).await;
-                            }
-                        }
-                    }
+            }
+            WalletConsensusItem::Feerate(feerate) => {
+                if Some(feerate) == dbtx.insert_entry(&FeeRateVoteKey(peer_id), &feerate).await {
+                    bail!("Fee rate vote is redundant");
+                }
+            }
+            WalletConsensusItem::PegOutSignature(peg_out_signature) => {
+                let txid = peg_out_signature.txid;
+                let unsigned_key = UnsignedTransactionKey(txid);
+
+                let mut unsigned = match dbtx.get_value(&unsigned_key).await {
+                    Some(unsigned) => unsigned,
+                    None => bail!("Unsigned trasnaction for this peg out signature does not exist"),
+                };
+
+                if self
+                    .sign_peg_out_psbt(&mut unsigned.psbt, &peer_id, &peg_out_signature)
+                    .is_err()
+                {
+                    bail!("Peg out signature is invalid");
+                }
```

Comment:
```suggestion
                self
                    .sign_peg_out_psbt(&mut unsigned.psbt, &peer_id, &peg_out_signature)
                    .context("Peg out signature is invalid")?;
```

Reply:
This is a good suggestion and has me actually questioning wether we should differentiate between invalid items we should log a warning for and merely redundant items in the returned result so we can log the warning at a higher level and use context as you described. converted the pr back to a draft to work this out...

Reply:
I have update the code to only return an error if we want it to be logged, otherwise we return an Ok(ConsensusDecision::Discard). Then the caller can log the error.